test/intersections.jl

```diff
@@ -130,6 +130,123 @@ using Base: hash_integer
     @test isa(intersecttype(s1, s12), CornerTouchingSegments) # CornerTouchingSegments (colinear)
     @test s1 ∩ s12 == P3(1.0, 0.0, 0.0)
   end
+    
+  @testset "Triangles" begin
+    # Utility to reverse segments, to more fully test branches in the intersection algorithm
+    reverse_segment(s) = Segment(vertices(s)[2], vertices(s)[1])
+
+    # segments and triangles in 3D
+    t₁ = Triangle(P3(0, 0, 0), P3(1, 0, 0), P3(0, 1, 0))  # Triangle lying in XY plane
+    t₂ = Triangle(P3(0, 0, 0), P3(2, 0, 0), P3(0, 2, 2))  # Inclined triangle
+    
+    ## Intersections with t₁
+    # Intersects through t₁
+    s₁ = Segment(P3(0.2, 0.2, 1.0), P3(0.2, 0.2, -1.0))
+    @test isa(intersecttype(s₁, t₁), IntersectingSegmentTriangle)
```

Comment:
Please use the infix version `intersecttype(s1, t1) isa IntersectingSegmentTriangle`